docs/generate-oval.py

```diff
@@ -0,0 +1,147 @@
+        "xmlns:red-def", "http://oval.mitre.org/XMLSchema/oval-definitions-5#linux"
+    )
+    oval_definitions.set(
+        "xmlns:ind-def",
+        "http://oval.mitre.org/XMLSchema/oval-definitions-5#independent",
+    )
+    oval_definitions.set("xmlns:xsi", "http://www.w3.org/2001/XMLSchema-instance")
+    oval_definitions.set(
+        "xsi:schemaLocation",
+        "http://oval.mitre.org/XMLSchema/oval-common-5 oval-common-schema.xsd http://oval.mitre.org/XMLSchema/oval-definitions-5 oval-definitions-schema.xsd http://oval.mitre.org/XMLSchema/oval-definitions-5#unix unix-definitions-schema.xsd http://oval.mitre.org/XMLSchema/oval-definitions-5#linux linux-definitions-schema.xsd",
+    )
+
+    generator = ET.SubElement(oval_definitions, "generator")
+    for key, value in {
+        "oval:product_name": "Red Hat OVAL Patch Definition Merger",
+        "oval:product_version": "3",
+        "oval:schema_version": "5.11.3",
+        "oval:timestamp": "2022-09-24T10:42:01",
+        "oval:content_version": "1664016121",
+    }.items():
+        oval = ET.SubElement(generator, key)
+        oval.text = value
+
+    definitions = ET.SubElement(oval_definitions, "definitions")
+    for item in items:
+
+        definition = ET.SubElement(definitions, "definition")
+        definition.set("class", "patch")
+        definition.set("id", "oval:{}:def:1".format(item["id"]))
+        definition.set("version", "1")
+
+        metadata = ET.SubElement(definition, "metadata")
+
+        ET.SubElement(metadata, "title").text = item["name"]
+
+        affected = ET.SubElement(metadata, "affected")
+        affected.set("family", "unix")
+        ET.SubElement(affected, "platform").text = "All"  # is this valid?
+
+        if "issues" in item:
+            for issue in item["issues"]:
+                reference = ET.SubElement(metadata, "reference")
+                reference.set("ref_id", issue)
+                if issue.startswith("CVE-"):
+                    reference.set(
+                        "ref_url", "https://nvd.nist.gov/vuln/detail/{}".format(issue)
+                    )
+                    reference.set("source", "CVE")
+
+        if "description" in item:
+            ET.SubElement(metadata, "description").text = "\n".join(item["description"])
+
+        criteria = ET.SubElement(definition, "criteria")
+        criteria.set("operator", "OR")
+        criterion = ET.SubElement(criteria, "criterion")
+        criterion.set("comment", item["name"])
+        criterion.set("test_ref", "oval:{}:tst:1".format(item["id"]))
+
+    tests = ET.SubElement(oval_definitions, "tests")
+    for item in items:
+
+        red_def = ET.SubElement(tests, "red-def:fwupdsecattr_test")
+        red_def.set("check", "at least one")
+        red_def.set("comment", item["name"])
+        red_def.set("id", "oval:{}:tst:1".format(item["id"]))
+        red_def.set("version", "1")
+
+        red_def_object = ET.SubElement(red_def, "red-def:object")
```

Comment:
The "object" in "tests" should be
<red-def:object **object_ref**="oval:org.fwupd.hsi.Amd.PlatformRollbackProtection:obj:1" />

not
<red-def:object **red-def:object**="oval:org.fwupd.hsi.Amd.PlatformRollbackProtection:obj:1" />

Reply:
Fixed, thanks.